dspace-server-webapp/src/test/java/org/dspace/app/rest/QAEventRestRepositoryIT.java

```diff
@@ -0,0 +1,790 @@
+                .andExpect(status().isUnprocessableEntity());
+        // check that no related item has been added to our event
+        getClient(authToken)
+            .perform(get("/api/integration/qualityassuranceevents/" + event.getEventId()).param("projection", "full"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", QAEventMatcher.matchQAEventFullEntry(event)));
+    }
+
+    @Test
+    public void deleteItemWithEventTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        parentCommunity = CommunityBuilder.createCommunity(context).withName("Parent Community").build();
+        Collection col1 = CollectionBuilder.createCollection(context, parentCommunity).withName("Collection 1").build();
+        QAEvent event1 = QAEventBuilder.createTarget(context, col1, "Science and Freedom")
+                .withTopic("ENRICH/MISSING/PID")
+                .withMessage("{\"pids[0].type\":\"doi\",\"pids[0].value\":\"10.2307/2144300\"}").build();
+        QAEvent event2 = QAEventBuilder.createTarget(context, col1, "Science and Freedom 2")
+                .withTopic("ENRICH/MISSING/PID")
+                .withMessage("{\"pids[0].type\":\"doi\",\"pids[0].value\":\"10.2307/2144301\"}").build();
+        context.restoreAuthSystemState();
+        String authToken = getAuthToken(admin.getEmail(), password);
+        getClient(authToken)
+            .perform(
+                get("/api/integration/qualityassuranceevents/search/findByTopic").param("topic", "ENRICH!MISSING!PID"))
+            .andExpect(status().isOk()).andExpect(jsonPath("$._embedded.qualityassuranceevents", Matchers.hasSize(2)))
+            .andExpect(jsonPath("$._embedded.qualityassuranceevents",
+                        Matchers.containsInAnyOrder(QAEventMatcher.matchQAEventEntry(event1),
+                                QAEventMatcher.matchQAEventEntry(event2))))
+                .andExpect(jsonPath("$.page.size", is(20))).andExpect(jsonPath("$.page.totalElements", is(2)));
+
+        getClient(authToken).perform(delete("/api/core/items/" + event1.getTarget()))
+                .andExpect(status().is(204));
+
+        getClient(authToken).perform(get("/api/core/items/" + event1.getTarget()))
+                .andExpect(status().is(404));
+
+        getClient(authToken)
+            .perform(
+                get("/api/integration/qualityassuranceevents/search/findByTopic").param("topic", "ENRICH!MISSING!PID"))
+            .andExpect(status().isOk()).andExpect(jsonPath("$._embedded.qualityassuranceevents", Matchers.hasSize(1)))
+            .andExpect(jsonPath("$._embedded.qualityassuranceevents",
+                        Matchers.containsInAnyOrder(
+                                QAEventMatcher.matchQAEventEntry(event2))))
+                .andExpect(jsonPath("$.page.size", is(20))).andExpect(jsonPath("$.page.totalElements", is(1)));
+    }
+
+    @Test
+    public void testEventDeletion() throws Exception {
```

Comment:
Per comment above about no permissions checking on deletion, I notice that we also don't have a test that proves non-admins cannot delete.